modules/core/src/main/java/org/apache/ignite/internal/managers/tracing/GridTracingManager.java

```diff
@@ -205,14 +219,14 @@ private Span enrichWithLocalNodeParameters(@Nullable Span span) {
         Span span;
 
         try {
-            if (serializedParentSpan == null || serializedParentSpan == NoopTracing.NOOP_SERIALIZED_SPAN)
+            if (serializedParentSpan == null || serializedParentSpan.length == 0)
                 return create(spanType, NoopSpan.INSTANCE);
 
             // First byte of the serializedSpan is reserved for special flags - it's not used right now.
 
             // Deserialize and compare spi types. If they don't match (span was serialized with another spi) then
             // propagate serializedSpan as DeferredSpan.
-            if (serializedParentSpan[SPI_TYPE_OFF] != getSpi().type().index())
+            if (serializedParentSpan[SPI_TYPE_OFF] != getSpi().type())
```

Comment:
Please take a look at `switch` below. Do we need to handle a minor version of the protocol that differs from 0?